project/grammars/cyk.py

```diff
@@ -0,0 +1,52 @@
+from pyformlang.cfg import CFG
+
+__all__ = ["cyk"]
+
+
+def cyk(cfg: CFG, word: str) -> bool:
+    """
+    Cocke-Younger-Kasami algorithm implementation.
+    Check whether CFG accepts given word
+
+    Parameters
+    ----------
+    cfg: CFG
+        Context Free Grammar
+    word: str
+        Word to check acceptance
+
+    Returns
+    -------
+    is_accepted: bool
+        True if CFG accepts word, False otherwise
+    """
+    word_len = len(word)
+
+    if not word_len:
+        return cfg.generate_epsilon()
+
+    cnf = cfg.to_normal_form()
+
+    term_productions = [p for p in cnf.productions if len(p.body) == 1]
+    var_productions = [p for p in cnf.productions if len(p.body) == 2]
+
+    m = [[set() for _ in range(word_len)]] * word_len
```

Comment:
```python
>>> d = [[set() for _ in range(2)]] * 2
>>> d
[[set(), set()], [set(), set()]]
>>> d[0][0].add(1)
>>> d
[[{1}, set()], [{1}, set()]]
```
Я думаю, вы такого не ожидали.

Лучше сделать так.
```suggestion
    m = [[set() for _ in range(word_len)] for _ in range(word_len)]
```

Тогда будет
```python
>>> d = [[set() for _ in range(2)] for _ in range(2)] 
>>> d
[[set(), set()], [set(), set()]]
>>> d[0][0].add(1)
>>> d
[[{1}, set()], [set(), set()]]
```

Reply:
Это действительно сильно влияет на корректность работы алгоритма. Исправлено.